### PR TITLE
Remove processSlots from slashing check

### PR DESCRIPTION
**Pull request by prestonvanloon**
**What type of PR is this?**

Other

**What does this PR do? Why is it needed?**

ProcessSlots was unnecessary and could cause undue burdon when processing slashing events.

**Which issues(s) does this PR fix?**

N/A

**Other notes for review**

### CODE REVIEW

File: beacon-chain/core/blocks/proposer_slashing.go [hunk 1]
```diff
@@ -86,7 +86,7 @@ func VerifyProposerSlashing(
 	if err != nil {
 		return err
 	}
-	if !helpers.IsSlashableValidatorUsingTrie(proposer, helpers.SlotToEpoch(beaconState.Slot())) {
+	if !helpers.IsSlashableValidatorUsingTrie(proposer, helpers.SlotToEpoch(hSlot)) {
```
Comment (nisdas) [inline beacon-chain/core/blocks/proposer_slashing.go:89]: can we add a regression test here.

**Review by nisdas — COMMENTED** [review]
Removing this becomes tricky when we hit a certain edge case, if we get a slashing after a forked epoch but our current state is before the forked epoch. Since we need the relevant fork version retrieved from the state to be accurate. 

I guess the chances of this happening is small and it will most likely be a unique edge case so maybe its not something to worry about.

@terencechain wdyt ?

**Comment by prestonvanloon** [comment]
> Removing this becomes tricky when we hit a certain edge case, if we get a slashing after a forked epoch but our current state is before the forked epoch. Since we need the relevant fork version retrieved from the state to be accurate.
>
> I guess the chances of this happening is small and it will most likely be a unique edge case so maybe its not something to worry about.
>
> @terencechain wdyt ?


@nisdas, that edge case would need to be handled in the forking logic. I outlined this issue in my [fork notes](https://hackmd.prylabs.network/s/BJ6-wdFHw) and I think the edge case you've highlighted is a greater concern than this PR specifically. In any case, we should use the slashing's slot to determine the fork version epoch to read from the state. So, the state's slot shouldn't matter given that we only need to read state.fork (which is not affected by processSlots).

File: beacon-chain/core/blocks/proposer_slashing.go (hunk 1 above)
Comment (prestonvanloon) [reply beacon-chain/core/blocks/proposer_slashing.go:89]: For sure, thanks!

**Review by nisdas — APPROVED** [review]
lgtm
